examples/trivial/main.rs

```diff
@@ -1,117 +1,57 @@
-const COMMITMENT_KEY_SIZE: usize = 20;
-const T: usize = 5;
-const RATE: usize = 4;
+/// Arity : Input/output size per fold-step for primary step-circuit
+/// For tivial case it can be any number
+const A1: usize = 5;
 
-type RandomOracle = poseidon::PoseidonRO<T, RATE>;
+/// Arity : Input/output size per fold-step for secondary step-circuit
+/// For tivial case it can be any number
+const A2: usize = 1;
 
-type RandomOracleConstant<F> = <RandomOracle as ROPair<F>>::Args;
+/// Table size for Primary Circuit
+///
+/// Requires at least 17, for service purposes, but if the primary requires more, increase the
+/// constant
+const SECONDARY_CIRCUIT_TABLE_SIZE: usize = 17;
 
-const LIMB_WIDTH: NonZeroUsize = unsafe { NonZeroUsize::new_unchecked(32) };
-const LIMBS_COUNT_LIMIT: NonZeroUsize = unsafe { NonZeroUsize::new_unchecked(10) };
+/// Key size for Primary Circuit
+const PRIMARY_COMMITMENT_KEY_SIZE: usize = 21;
 
-type C1Affine = <C1 as PrimeCurve>::Affine;
-type C2Affine = <C2 as PrimeCurve>::Affine;
+/// Table size for Primary Circuit
+///
+/// Requires at least 17, for service purposes, but if the primary requires more, increase the
+/// constant
+const PRIMARY_CIRCUIT_TABLE_SIZE: usize = 17;
 
-type C1Scalar = <C1 as Group>::Scalar;
-type C2Scalar = <C2 as Group>::Scalar;
+/// Key size for Primary Circuit
+const SECONDARY_COMMITMENT_KEY_SIZE: usize = 21;
 
-const FOLDER: &str = ".cache/examples";
-
-#[instrument]
-fn get_or_create_commitment_key<C: CurveAffine>(
-    k: usize,
-    label: &'static str,
-) -> io::Result<CommitmentKey<C>> {
-    unsafe { CommitmentKey::load_or_setup_cache(Path::new(FOLDER), label, k) }
-}
+use sirius::prelude::bn256::{new_default_pp, C1Affine, C1Scalar, C2Affine, C2Scalar};
 
 fn main() {
-    let builder = tracing_subscriber::fmt()
-        .with_span_events(FmtSpan::ENTER | FmtSpan::CLOSE)
-        .with_env_filter(
-            EnvFilter::builder()
-                .with_default_directive(LevelFilter::INFO.into())
-                .from_env_lossy(),
-        );
-
-    if env::args().any(|arg| arg.eq("--json")) {
-        builder.json().init();
-    } else {
-        builder.init();
-    }
-
-    let _span = info_span!("trivial_example").entered();
-
-    // C1
-    let sc1 = step_circuit::trivial::Circuit::<ARITY, _>::default();
-    // C2
-    let sc2 = step_circuit::trivial::Circuit::<ARITY, _>::default();
-
-    let primary_spec = RandomOracleConstant::<<C1 as CurveExt>::ScalarExt>::new(10, 10);
-    let secondary_spec = RandomOracleConstant::<<C2 as CurveExt>::ScalarExt>::new(10, 10);
+    let sc1 = trivial::Circuit::<A1, C1Scalar>::default();
+    let sc2 = trivial::Circuit::<A2, C2Scalar>::default();
 
     let primary_commitment_key =
-        get_or_create_commitment_key::<C1Affine>(COMMITMENT_KEY_SIZE, "bn256")
-            .expect("Failed to get primary key");
-    info!("Primary generated");
+        CommitmentKey::<C1Affine>::setup(PRIMARY_COMMITMENT_KEY_SIZE, b"bn256");
```

Comment:
Does this "setup" method serve the same function as "get_or_create"? i.e. can it get an existing key?

Reply:
No, this PR disables caching at this point for this example. It overcomplicates the example

Reply:
There is no difference from user's point of view.  Actually, it is an important feature for user experience. When run the second time, it will be much faster. Even if you use setup here, we can still keep the original function for internal ussage.

Reply:
Every line of the example should be clear and easy to understand. Choosing a folder for caching (even unsafe) - already increases the cognitive load for the startup. That's why I want to put it in a separate documentation item. For the trivial example, a key of size 21 is generated fast enough to favor a reduced cognitive load over the startup time of the example

And so this is of course an important functionality, it should be reflected, just not here. And it still needs to be improved by adding a choice of cache folder from the git repository rather than the startup directory